Build failure: ntopng

### Nixpkgs version

- Unstable (25.05)

### Steps to reproduce

1. Checkout recent master commit (I used `58edd1e2acbc9be9fe29964344c6419db013141e`)
2. `nix-build -A ntopng`
3. See build failure


### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/287883380

### Relevant log output

```console
make: *** [Makefile:133: src/Flow.o] Error 1
error: builder for '/nix/store/5ah50rsnyib4cif4r1ss4hasrcrr5prn-ntopng-6.2.drv' failed with exit code 2;
       last 25 log lines:
       > src/Flow.cpp:6883:88: error: 'const ndpi_protocol' {aka 'const struct ndpi_proto'} has no member named 'master_protocol'
       >  6883 |                                 (!ndpi_is_subprotocol_informative(ndpiDetectedProtocol.master_protocol)
       >       |                                                                                        ^~~~~~~~~~~~~~~
       > src/Flow.cpp:6884:57: error: 'const ndpi_protocol' {aka 'const struct ndpi_proto'} has no member named 'app_protocol'
       >  6884 |                                  ? ndpiDetectedProtocol.app_protocol
       >       |                                                         ^~~~~~~~~~~~
       > src/Flow.cpp:6885:57: error: 'const ndpi_protocol' {aka 'const struct ndpi_proto'} has no member named 'master_protocol'
       >  6885 |                                  : ndpiDetectedProtocol.master_protocol));
       >       |                                                         ^~~~~~~~~~~~~~~
       > src/Flow.cpp:6887:54: error: 'const ndpi_protocol' {aka 'const struct ndpi_proto'} has no member named 'master_protocol'
       >  6887 |                                 ndpiDetectedProtocol.master_protocol);
       >       |                                                      ^~~~~~~~~~~~~~~
       > src/Flow.cpp: In member function 'void Flow::lua_get_min_info(lua_State*)':
       > src/Flow.cpp:7159:69: error: 'ndpi_protocol' {aka 'struct ndpi_proto'} has no member named 'app_protocol'
       >  7159 |                                                ndpiDetectedProtocol.app_protocol));
       >       |                                                                     ^~~~~~~~~~~~
       > src/Flow.cpp: In member function 'void Flow::getInfo(ndpi_serializer*)':
       > src/Flow.cpp:7201:73: error: 'ndpi_protocol' {aka 'struct ndpi_proto'} has no member named 'app_protocol'
       >  7201 |                                                    ndpiDetectedProtocol.app_protocol));
       >       |                                                                         ^~~~~~~~~~~~
       > src/Flow.cpp: In member function 'void Flow::updateUDPHostServices()':
       > src/Flow.cpp:8511:30: error: 'ndpi_protocol' {aka 'struct ndpi_proto'} has no member named 'app_protocol'
       >  8511 |     if((ndpiDetectedProtocol.app_protocol == NDPI_PROTOCOL_DOH_DOT)
       >       |                              ^~~~~~~~~~~~
       > make: *** [Makefile:133: src/Flow.o] Error 1
```

### Additional context

Attached is full build log: [ntopng-master.log](https://github.com/user-attachments/files/18732494/ntopng-master.log)

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.60, NixOS, 24.11 (Vicuna), 24.11pre-git`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.24.10`
 - nixpkgs: `/nix/store/4r8s42c9mwfvgdlxv0izb1cmlzrsb5nz-source`



### Notify maintainers

@bjornfor 
---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/nd/ndpi/package.nix
+++ b/pkgs/by-name/nd/ndpi/package.nix
@@ -12,13 +12,13 @@
 
 stdenv.mkDerivation (finalAttrs: {
   pname = "ndpi";
-  version = "4.12";
+  version = "4.10";
 
   src = fetchFromGitHub {
     owner = "ntop";
     repo = "nDPI";
     tag = finalAttrs.version;
-    hash = "sha256-jdyKvM/Tb9pRWQPxpB/UQOOKamWrTS24Ofc3M5M1Zso=";
+    hash = "sha256-iXqvDMJsOXcg9YkqKFgInLLfH6j/HEp4bEaIl6dpVtc=";
   };
 
   nativeBuildInputs = [

--- a/pkgs/by-name/nt/ntopng/package.nix
+++ b/pkgs/by-name/nt/ntopng/package.nix
@@ -81,6 +81,16 @@ stdenv.mkDerivation (finalAttrs: {
         -i include/ntop_defines.h
   '';
 
+  # Upstream build system makes
+  # $out/share/ntopng/httpdocs/geoip/README.geolocation.md a dangling symlink
+  # to ../../doc/README.geolocation.md. Copying the whole doc/ tree adds over
+  # 70 MiB to the output size, so only copy the files we need for now.
+  # (Ref. noBrokenSymlinks.)
+  postInstall = ''
+    mkdir -p "$out/share/ntopng/doc"
+    cp -r doc/README.geolocation.md "$out/share/ntopng/doc/"
+  '';
+
   enableParallelBuilding = true;
 
   meta = with lib; {

